x-pack/qa/security-example-spi-extension/src/test/java/org/elasticsearch/example/realm/CustomRealmTests.java

```diff
@@ -17,26 +18,56 @@
 import org.elasticsearch.xpack.core.security.authc.support.UsernamePasswordToken;
 import org.elasticsearch.xpack.core.security.user.User;
 
+import java.util.List;
+
 import static org.elasticsearch.xpack.core.security.authc.RealmSettings.getFullSettingKey;
+import static org.hamcrest.Matchers.arrayContaining;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.notNullValue;
 
 public class CustomRealmTests extends ESTestCase {
-    public void testAuthenticate() {
+
+    public void testAuthenticateDefaultConfig() {
         Settings globalSettings = Settings.builder().put("path.home", createTempDir()).build();
         final RealmConfig.RealmIdentifier realmIdentifier = new RealmConfig.RealmIdentifier(CustomRealm.TYPE, "test");
         CustomRealm realm = new CustomRealm(new RealmConfig(
             realmIdentifier,
             Settings.builder().put(globalSettings).put(getFullSettingKey(realmIdentifier, RealmSettings.ORDER_SETTING), 0).build(),
             TestEnvironment.newEnvironment(globalSettings), new ThreadContext(globalSettings)));
-        SecureString password = CustomRealm.KNOWN_PW.clone();
-        UsernamePasswordToken token = new UsernamePasswordToken(CustomRealm.KNOWN_USER, password);
+        SecureString password = CustomRealm.DEFAULT_KNOWN_PW.clone();
+        UsernamePasswordToken token = new UsernamePasswordToken(CustomRealm.DEFAULT_KNOWN_USER, password);
+        PlainActionFuture<AuthenticationResult> plainActionFuture = new PlainActionFuture<>();
+        realm.authenticate(token, plainActionFuture);
+        User user = plainActionFuture.actionGet().getUser();
+        assertThat(user, notNullValue());
+        assertThat(List.of(user.roles()), equalTo(CustomRealm.DEFAULT_ROLES));
+        assertThat(user.principal(), equalTo(CustomRealm.DEFAULT_KNOWN_USER));
+    }
+
+    public void testAuthenticateCustomConfig() {
+        final RealmConfig.RealmIdentifier realmIdentifier = new RealmConfig.RealmIdentifier(CustomRealm.TYPE, "test");
+        MockSecureSettings secureSettings = new MockSecureSettings();
+        secureSettings.setString(getFullSettingKey(realmIdentifier.getName(), CustomRealm.PASSWORD_SETTING), "12345");
```

Comment:
Unless you are intentionally showing that password here can bypass the "at last 6-chars" policy, I'd prefer it to be compliant for consistency.

Reply:
I'm disappointed that my film references are being lost here, but will do.